### PR TITLE
update supported OSes in params.pp

### LINKED ISSUE #291: debian 7 support broken
## Affected Puppet, Ruby, OS and module versions/distributions

operating system debian 7

## How to reproduce (e.g Puppet code you use)

```
      openvpn::server { 'test_openvpn_server':
        country      => 'FR',
        province     => 'Occitanie',
        city         => 'France',
        organization => 'FOO',
        email        => 'bar@foo.org',
        server       => '10.0.0.0 255.255.255.0',
        local        => '',
      }
```

## What are you seeing

When this module is used on debian 7, the current settings about cipher with a so old operating system, does not work. The service `openvpn` does not start

## What behaviour did you expect instead

process `openvpn` started.

## Output log

The start of openvpn server log the following : 
`Problem with cipher list: TLS-DHE-RSA-WITH-AES-256-GCM-SHA384:TLS-DHE-RSA-WITH-AES-256-CBC-SHA256:TLS-DHE-RSA-WITH-AES-128-GCM-SHA256:TLS-DHE-RSA-WITH-AES-128-CBC-SHA256: error:1410D0B9:SSL routines:SSL_CTX_set_cipher_list:no cipher match`

If removing debian 7 support is considered as possible option,  i can provide a PR.

### PR DESCRIPTION
**Pull request by Dan33l**
#### Pull Request (PR) description

update `params.pp` to be up to date with current `metadata.json`

#### This Pull Request (PR) fixes the following issues

Fixes #291

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -16,98 +16,71 @@
 #
 class openvpn::params {
 
-  case $::osfamily {
-    'RedHat': {
+  case $facts['os']['family'] {
```
Comment (bastelfreak) [inline manifests/params.pp]: structured facts \o/

File: manifests/params.pp [hunk 2]
```diff
@@ -16,98 +16,71 @@
...
+      case $facts['os']['release']['major'] {
+        '7': {
+          $additional_packages = ['easy-rsa']
+          $ldap_auth_plugin_location = undef
+          $systemd = true
+        }
+        '6': {
+          $additional_packages = ['easy-rsa','openvpn-auth-ldap']
+          $ldap_auth_plugin_location = '/usr/lib64/openvpn/plugin/lib/openvpn-auth-ldap.so'
+          $systemd = false
```
Comment (bastelfreak) [inline manifests/params.pp]: suggestion for future enhancements: ```$systemd``` can be set to ```true``` if the fact ```[service_provider```](https://github.com/puppetlabs/puppetlabs-stdlib#service_provider) returns ```systemd```

File: spec/classes/openvpn_config_spec.rb [hunk 3]
```diff
@@ -4,9 +4,9 @@
   context 'on Debian based machines' do
     let(:facts) do
       {
-        osfamily: 'Debian',
-        operatingsystem: 'Debian',
-        operatingsystemrelease: '7',
+        os: { 'family' => 'Debian' },
```
Comment (bastelfreak) [inline spec/classes/openvpn_config_spec.rb]: this should be migrated to rspec-puppet-facts, same applies for the other spec files.

Comment (Dan33l) [reply spec/classes/openvpn_config_spec.rb]: You're right, but the tag `needs-work` was not useless.
It did it.

File: spec/classes/openvpn_config_spec.rb [hunk 4]
```diff
@@ -1,37 +1,41 @@
...
-        operatingsystem: 'Debian',
-        operatingsystemrelease: '7',
-        concat_basedir: '/var/lib/puppet/concat'
-      }
-    end
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts.merge(
+          concat_basedir: '/var/lib/puppet/concat'
```
Comment (bastelfreak) [inline spec/classes/openvpn_config_spec.rb]: Did you test if this is required? We have mocked facts here: https://github.com/voxpupuli/puppet-openvpn/blob/9f3abf684f9d817e67a56469c15ea4028eae34d7/spec/default_facts.yml#L11

And use them here:
https://github.com/voxpupuli/puppet-openvpn/blob/9f3abf684f9d817e67a56469c15ea4028eae34d7/spec/spec_helper.rb#L25-L35

**Comment by bastelfreak** [comment]
If I see this correctly this PR doesnt add support for new operating systems, right? Otherwise the list in the metadata.json needs to be updated as well.

**Comment by bastelfreak** [comment]
I created a little patch that you might want to look into:

```patch
diff --git a/spec/classes/openvpn_config_spec.rb b/spec/classes/openvpn_config_spec.rb
index 10b4e73..d33e35d 100644
--- a/spec/classes/openvpn_config_spec.rb
+++ b/spec/classes/openvpn_config_spec.rb
@@ -4,12 +4,12 @@ describe 'openvpn::config', type: :class do
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       let(:facts) do
-        facts.merge(
-          concat_basedir: '/var/lib/puppet/concat'
-        )
+        facts
       end
 
-      case facts[:os][:family]
+      it { is_expected.to compile.with_all_deps }
+
+      case facts[:os]['family']
       when 'Debian'
         context 'on Debian based machines' do
           it { is_expected.to contain_concat('/etc/default/openvpn') }
```

A couple of things:
* I tested this, we don't need to mock concat_basedir
* in the facts hash, only the first key is a symbol, all further nested keys are strings
* It's always good to check if the manifest compiles into a catalog

**Comment by Dan33l** [comment]
> If I see this correctly this PR doesnt add support for new operating systems, right? Otherwise the list in the metadata.json needs to be updated as well.

Exactly . No new OS.

File: manifests/server.pp [hunk 5]
```diff
@@ -454,7 +454,7 @@
   Optional[String] $group                                           = undef,
   Boolean $ipp                                                      = false,
   Boolean $duplicate_cn                                             = false,
-  String $local                                                     = $::ipaddress_eth0,
+  String $local                                                     = $facts['ipaddress_eth0'],
```
Comment (bastelfreak) [inline manifests/server.pp]: facts hash \o/

**Comment by bastelfreak** [comment]
thanks for this awesome PR!

**Pull request merged**